lib/src/Dropdown/Dropdown.tsx

```diff
@@ -79,14 +79,32 @@ export const Dropdown: React.FC<DropdownProps> = React.forwardRef(({ wrapperProp
         [isMobile, props.multiple, onMultipleChange]
     );
 
-    const selectAll = (forceValue?: boolean | React.ChangeEvent<HTMLInputElement>) => {
-        Array.from(selectRef.current.options).forEach((option) => {
-            option.selected = typeof forceValue === "boolean" ? forceValue : !allSelected;
+    const selectAll = React.useCallback(
+        (forceValue?: boolean | React.ChangeEvent<HTMLInputElement>) => {
+            Array.from(selectRef.current.options).forEach((_, i) => {
+                const option = selectRef.current.options.item(i);
+                if (!option.disabled) {
+                    option.selected = typeof forceValue === "boolean" ? forceValue : !allSelected;
+                } else {
+                    option.selected = false;
+                }
+            });
+            typeof forceValue === "boolean" && (selectRef.current.value = "");
+            selectRef.current.dispatchEvent(new Event("change", { bubbles: true }));
+            props.multiple && onMultipleChange && onMultipleChange(getValueOfMultipleSelect(selectRef.current));
+        },
+        [allSelected, props.multiple]
+    );
+
+    const isAllSelected = (): boolean => {
+        return Array.from(selectRef.current.options).every((_, i) => {
+            const option = selectRef.current.options.item(i);
+            if (option.disabled) {
+                return true;
+            } else {
+                return option.selected;
+            }
```

Comment:
```suggestion
          option.disabled ? true : option.selected
```

Reply:
do we need that true? option.disabled || option.selected